lib/ood_core/job/adapters/slurm.rb

```diff
@@ -11,16 +11,26 @@ class Factory
       # @option config [Object] :cluster (nil) The cluster to communicate with
       # @option config [Object] :conf (nil) Path to the slurm conf
       # @option config [Object] :bin (nil) Path to slurm client binaries
-      # @option config [Object] :major_version (17) Slurm's major version
+      # @option config [Object] :version (17) Slurm's version
       def self.build_slurm(config)
         c = config.to_h.symbolize_keys
         cluster = c.fetch(:cluster, nil)
         conf    = c.fetch(:conf, nil)
         bin     = c.fetch(:bin, nil)
-        major_version = c.fetch(:major_version, 17)
+        major_version = parse_major_version(c.fetch(:version, nil))
+
+        raise 'OodCore::Job::Adapters::Slurm supports Slurm version 17 and greater' if major_version == 0
+        
         slurm = Adapters::Slurm::Batch.new(cluster: cluster, conf: conf, bin: bin, major_version: major_version)
         Adapters::Slurm.new(slurm: slurm)
       end
+
+      def parse_major_version(version)
+        # Backward compatible behavior for current installations
+        return 17 if version.nil?
+
+        version.to_i
+      end
```

Comment:
Why do `major_version = parse_major_version(c.fetch(:version, nil))` when the same functionality can be done by just `major_version = c.fetch(:major_version, 17).to_i`?

Reply:
Also note that this block of code is actually opening OodCore::Job::Factory and adding the method `build_slurm` and `parse_major_version`. At least `build_slurm` has somewhat of a "namespace" with the suffix, but the `parse_major_version` is probably too generic of a method to be added to OodCore::Job::Factory. Imagine if we took the same approach with another adapter wanted to implement a similar method - one implementation would replace the other introducing a defect because we would now have two implementations of `OodCore::Job::Factory#parse_major_version`.

Reply:
See https://github.com/OSC/ood_core/issues/89 - I want to simplify the creation of these adapter instances. The result would be removing or moving the build methods to the Adapter classes themselves. 

If you are curious why the original complexity exists, we first copied what Rails does with loading database adapters since. But YAGNI.